src/improvements/template/OPCMUpgradeV200.sol

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {OPCMBaseTask} from "../tasks/OPCMBaseTask.sol";
+import {SuperchainAddressRegistry} from "src/improvements/SuperchainAddressRegistry.sol";
+import {
+    IOPContractsManager,
+    ISystemConfig,
+    IProxyAdmin
+} from "@eth-optimism-bedrock/interfaces/L1/IOPContractsManager.sol";
+import {IStandardValidatorV200} from "@eth-optimism-bedrock/interfaces/L1/IStandardValidator.sol";
+import {Claim} from "@eth-optimism-bedrock/src/dispute/lib/Types.sol";
+import {VmSafe} from "forge-std/Vm.sol";
+
+/// @notice This template supports OPCMV200 upgrade tasks.
+contract OPCMUpgradeV200 is OPCMBaseTask {
+    /// @notice The OPContractsManager address
+    address public OPCM;
+
+    /// @notice The StandardValidatorV200 address
+    IStandardValidatorV200 public STANDARD_VALIDATOR_V200;
+
+    /// @notice Struct to store inputs for OPCM.upgrade() function per l2 chain
+    struct OPCMUpgrade {
+        Claim absolutePrestate;
+        uint256 chainId;
+    }
+
+    /// @notice Mapping of l2 chain IDs to their respective prestates
+    mapping(uint256 => Claim) public opcmUpgrades;
+
+    /// @notice Returns the OPCM address
+    function opcm() public view override returns (address) {
+        require(OPCM != address(0), "OPCMUpgradeV200: OPCM address not set in template");
+        return OPCM;
+    }
+
+    /// @notice Returns the storage write permissions
+    function _taskStorageWrites() internal pure virtual override returns (string[] memory) {
+        string[] memory storageWrites = new string[](13);
+
+        // Upgraded contracts
+        storageWrites[0] = "OPContractsManager";
```

Comment:
I would personally like if we had access to the address registry in here. This would be we could find the `PermissionedWETH` and `PermissionlessWETH` addresses instead of having to provide them in the config.toml. The array would become: 
```
address[] memory storageWrites = new address[](13);
```

This can be done in a later PR.